### PR TITLE
[types/ember__array] Fix type for `compact`, `firstObject` and `lastObject`

### PR DESCRIPTION
**Pull request by lolmaus**
Adresses https://github.com/typed-ember/ember-cli-typescript/issues/1145

Does this need a test?  I honestly tried to do it, but failed to create a failing test. This dtslint thing just won't produce the error that normal TypeScript [produces](https://www.typescriptlang.org/play/#)

### CODE REVIEW

**Comment by lolmaus** [comment]
CI failures are unrelated to my one-liner change.

**Comment by dfreeman** [comment]
@lolmaus one of the errors does look related to your change: https://travis-ci.org/github/DefinitelyTyped/DefinitelyTyped/builds/678764101#L335-L340

The others are addressed in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44159

**Comment by lolmaus** [comment]
@dfreeman, thanks for pointing out that file!

It is *not* run when testing locally with `npm lint ember__array` as the readme suggests. The documentation on testing is quite poor in the DefinitelyTyped readme. 😔

How do I run the tests? Every time I run `yarn test`, demands that I commit my changes, then it rebuilds the test pipeline. Is there a quicker was similar to `ember test --serve`?

***

Now I'm able to make a failing test case for my change, but I have the opposite problem: the test case won't go green after applying my fix.

Here's what I see in my app:

```ts
compact(): Array<NonNullable<T>>;
```

image

In other words, I expect `NonNullable` to remove `null` and `undefined` from the type.

But in DefinitelyTyped, I see the following:

```ts
compact(): NativeArray<NonNullable<T>>;
```

```ts
let me1: Ember.MutableEnumerable<[string | null | undefined]> = null as any;
me1.compact(); // $ExpectType NativeArray<[string]>
```

```
Error in ember
Error: C:/Code/DefinitelyTyped/types/ember/test/ember-module-tests.ts:232:1
ERROR: 232:1  expect  TypeScript@3.9 expected type to be:
  NativeArray<[string]>
got:
  NativeArray<[string | null | undefined]>
```

Please help.

**Comment by dfreeman** [comment]
You have extra brackets around your type; you want `<string | null | undefined>` without the `[]`s.

**Comment by jamescdavis** [comment]
@lolmaus maybe you forgot the `run`? `npm run lint ember__array` (I do this often when switching back to npm from yarn)

**Review by jamescdavis — CHANGES_REQUESTED** [review]
Oops, looks like you accidentally pulled in `types/trtc-js-sdk`

**Comment by lolmaus** [comment]
> Oops, looks like you accidentally pulled in types/trtc-js-sdk

Yeah, off-by-one error when squashing. I hate rebase. 😬 

All fixed now!

File: types/ember/test/ember-module-tests.ts [hunk 1]
```diff
@@ -228,7 +228,7 @@ ma1.addObject('!'); // $ExpectType string
 ma1.filterBy(''); // $ExpectType NativeArray<string>
 // Ember.MutableEnumerable
 // tslint:disable-next-line:prefer-const
-let me1: Ember.MutableEnumerable<[string]> = null as any;
+let me1: Ember.MutableEnumerable<[string] | null | undefined> = null as any;
 me1.compact(); // $ExpectType NativeArray<[string]>
```
Comment (jamescdavis) [inline types/ember/test/ember-module-tests.ts:232]: This test was weird to begin with, IMHO, and could be improved to:
```suggestion
let me1: Ember.MutableEnumerable<string | null | undefined> = ['foo', null, undefined];
me1.compact(); // $ExpectType NativeArray<string>
```

**Review by jamescdavis — CHANGES_REQUESTED** [review]

File: types/ember/test/ember-module-tests.ts (hunk 1 above)
Comment (jamescdavis) [reply types/ember/test/ember-module-tests.ts:232]: I'm also not sure why it uses `let` instead of `const`? Maybe to hint at mutability? IMHO it could be confusing since it's mutable regardless of `let` or `const`.

Comment (lolmaus) [reply types/ember/test/ember-module-tests.ts]: Done.

**Comment by lolmaus** [comment]
Also fixed types of `firstObject` and `lastObject`.

**Comment by lolmaus** [comment]
@jamescdavis @dfreeman Should be good now, please re-review.

How do I make the bot retry and update tags?

**Review by jamescdavis — APPROVED** [review]

**Comment by jamescdavis** [comment]
Thank you @lolmaus!
